packages/material-ui/src/ImageListItem/ImageListItem.js

```diff
@@ -1,128 +1,103 @@
+  // TODO: - Use jsdoc @default?: "cols rows default values are for docs only"
   const {
     children,
     classes,
     className,
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     cols = 1,
     component: Component = 'li',
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     rows = 1,
+    style,
     ...other
   } = props;
 
-  const imgRef = React.useRef(null);
-
-  React.useEffect(() => {
-    const img = imgRef.current;
-
-    if (!img) {
-      return undefined;
-    }
-
-    let listener;
-
-    if (img.complete) {
-      fit(img, classes);
-    } else {
-      listener = () => {
-        fit(img, classes);
-      };
-      img.addEventListener('load', listener);
-    }
-
-    return () => {
-      if (listener) {
-        img.removeEventListener('load', listener);
-      }
-    };
-  });
+  const { rowHeight = 'auto', spacing, variant } = React.useContext(ImageListContext);
 
-  React.useEffect(() => {
-    const handleResize = debounce(() => {
-      fit(imgRef.current, classes);
-    });
-
-    const containerWindow = ownerWindow(imgRef.current);
-    containerWindow.addEventListener('resize', handleResize);
-    return () => {
-      handleResize.clear();
-      containerWindow.removeEventListener('resize', handleResize);
-    };
-  }, [classes]);
+  let height = 'auto';
+  if (variant === 'woven') {
+    height = undefined;
+  } else if (rowHeight !== 'auto') {
+    height = rowHeight * rows + spacing * (rows - 1);
+  }
 
   return (
-    <Component className={clsx(classes.root, className)} ref={ref} {...other}>
-      <div className={classes.tile}>
-        {React.Children.map(children, (child) => {
-          if (!React.isValidElement(child)) {
-            return null;
+    <Component
+      className={clsx(classes.root, classes[variant], className)}
+      ref={ref}
+      style={{
+        height,
+        gridColumnEnd: variant !== 'masonry' ? `span ${cols}` : undefined,
+        gridRowEnd: variant !== 'masonry' ? `span ${rows}` : undefined,
+        marginBottom: variant === 'masonry' ? spacing : undefined,
+        ...style,
+      }}
+      {...other}
+    >
+      {React.Children.map(children, (child) => {
+        if (!React.isValidElement(child)) {
+          return null;
+        }
+
+        if (process.env.NODE_ENV !== 'production') {
+          if (isFragment(child)) {
+            console.error(
+              [
+                "Material-UI: The ImageListItem component doesn't accept a Fragment as a child.",
+                'Consider providing an array instead.',
+              ].join('\n'),
+            );
           }
+        }
 
-          if (child.type === 'img' || isMuiElement(child, ['Image'])) {
-            return React.cloneElement(child, {
-              ref: imgRef,
-            });
-          }
+        if (child.type === 'img' || isMuiElement(child, ['Image'])) {
+          return React.cloneElement(child, {
+            className: classes.img,
```

Comment:
Merge the class name?
```suggestion
            className: clsx(classes.img, child.props.className),
```